test/Constraints/generics.swift

```diff
@@ -251,7 +251,10 @@ struct V27515965 {
 }
 
 func test(x: S27515965) -> V27515965 {
-  return V27515965(x) // expected-error {{generic parameter 'T' could not be inferred}}
+  return V27515965(x)
+  // expected-error@-1 {{cannot invoke initializer for type 'init(_:)' with an argument list of type '(S27515965)'}}
+  // expected-error@-2 {{'<T where T : P27515965, T.R == Float> (V27515965.Type) -> (T) -> V27515965' requires the types 'Any' and 'Float' be equivalent}}
```

Comment:
Shouldn't this second error actually be a note? It's clarifying the previous error. Also, we shouldn't be printing the generic signature here. We'd like to point at the candidate and say something that describes the problem and how we got there within just one note (so we don't have the third diagnostic here):

    candidate requires that the types 'Any' and 'Float' be equivalent (i.e., 'T.R' == 'Float' with T == S27515965)

that approach scales to multiple candidates.